bleep-model/src/scala/bleep/model/ScriptDef.scala

```diff
@@ -1,21 +1,42 @@
 package bleep.model
 
-import io.circe.{Decoder, DecodingFailure, Encoder, Json}
+import io.circe.generic.semiauto.deriveCodec
+import io.circe._
 
-case class ScriptDef(project: CrossProjectName, main: String)
+sealed trait ScriptDef
 
 object ScriptDef {
-  implicit val decodes: Decoder[ScriptDef] = Decoder.instance(c =>
-    c.as[String].flatMap { str =>
-      str.split("/") match {
-        case Array(projectName, main) =>
-          CrossProjectName.decodes.decodeJson(Json.fromString(projectName)).map(crossProjectName => ScriptDef(crossProjectName, main))
-
-        case _ =>
-          Left(DecodingFailure(s"$str needs to be on the form `projectName(@crossId)/fully.qualified.Main`", c.history))
+  case class Main(project: CrossProjectName, main: String) extends ScriptDef
+  object Main {
+    implicit val codec: Codec.AsObject[Main] = deriveCodec
+  }
+
+  case class Shell(command: Option[String], `override-os`: Option[Map[String, String]]) extends ScriptDef
```

Comment:
What's the harm in making `command` mandatory?

Reply:
maybe there is value in writing a script which only works on one OS. not really sure. we can always constrain it later